### PR TITLE
enable passing proxy

### PR DESCRIPTION
**Pull request by jorgerosal**
Enables users to pass the proxy used from the first request.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ module.exports = (response) => {
 	const contentType = response.headers['content-type'];
 	const isHtml = contentType && contentType.split(';')[0] === 'text/html';
 	if (isHtml) {
-		return getPhantomHtml(response.request.href);
+		return getPhantomHtml(response.request.href, response.request.proxy);
```
Comment (s0ph1e) [inline index.js:15]: What happens in there is no `proxy` in request?

File: src/get-phantom-html.js [hunk 2]
```diff
@@ -6,9 +6,13 @@ const Promise = require('bluebird');
 
 const scriptPath = path.join(__dirname, 'script.js');
 
-module.exports = (url) => {
+module.exports = (url, proxy) => {
+	console.log(proxy.hostname);
+	var host = proxy.hostname;
+	var port = proxy.port;
```
Comment (s0ph1e) [inline src/get-phantom-html.js:12]: I think we need to check that `proxy` exists on line 11 and 12

and remove console.log

File: src/script.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
 var page = require('webpage').create();
 var system = require('system');
 
+
+phantom.setProxy(system.args[2], system.args[3]) //setProxy('ip','port')
```
Comment (s0ph1e) [inline src/script.js:5]: Do we need to check if `system.args[2]` and `system.args[3]` exist before calling setProxy?

**Review by s0ph1e — COMMENTED** [review]
Hi @yortrosal 

Thanks for your PR

I like the idea of proxy 
but let's ensure it will continue work without proxy in options and do not break something - I've added few comments about it

Also I suggest to remove commented out code and console logs

File: index.js (hunk 1 above)
Comment (jorgerosal) [reply index.js:15]: Yes, you're right. I have to admit, I wasn't really able to check if it still works without proxy in options. Let me do some test, make some changes and PR back. Thanks.

**Pull request closed without merging**